nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/FlattenJson.java

```diff
@@ -157,25 +191,35 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         final String mode = context.getProperty(FLATTEN_MODE).getValue();
         final FlattenMode flattenMode = getFlattenMode(mode);
 
-        String separator = context.getProperty(SEPARATOR).evaluateAttributeExpressions(flowFile).getValue();
-
+        final String separator = context.getProperty(SEPARATOR).evaluateAttributeExpressions(flowFile).getValue();
+        final String returnType = context.getProperty(RETURN_TYPE).getValue();
+        final PrintMode printMode = context.getProperty(PRETTY_PRINT).asBoolean() ? PrintMode.PRETTY : PrintMode.MINIMAL;
 
         try {
-            ByteArrayOutputStream bos = new ByteArrayOutputStream();
-            session.exportTo(flowFile, bos);
-            bos.close();
-
-            String raw = new String(bos.toByteArray());
-            final String flattened = new JsonFlattener(raw)
-                    .withFlattenMode(flattenMode)
-                    .withSeparator(separator.charAt(0))
-                    .withStringEscapePolicy(() -> StringEscapeUtils.ESCAPE_JSON)
-                    .flatten();
-
-            flowFile = session.write(flowFile, os -> os.write(flattened.getBytes()));
+            final StringBuilder contents = new StringBuilder();
+            session.read(flowFile, in -> contents.append(IOUtils.toString(in, Charset.defaultCharset())));
```

Comment:
Although the previous approach relied on the system default character set when converting from byte array to string, what do you think about either making UTF-8 the standard character set, or adding a new processor property to configure the character set?

Reply:
Yeah, let me add new character set property to stay consistent with other processors. Thanks.